### ISSUE TITLE
Receiving warning: oasOpParams expects a resolved object

### ISSUE THREAD

**Issue by cbetta** [stoplightio/spectral#88]
### What is the current behavior?

I am currently migrating my API specification and running into this warning:

```
oasOpParams expects a resolved object, but none was provided. Results may not be correct.
oasPathParam expects a resolved object, but none was provided. Results may not be correct.
```

There are no `$ref` mentions in my spec, so I am confused as to why I am getting this error. The code is as follows.

### What is the expected behavior?

For this not to error

### Please tell us about your environment:

  - Version:   `@stoplight/spectral": "^1.0.1"`
  - Framework: Node v11.7.0
  - Language: Node v11.7.0

### Other information

My code:

```js
// Load Spectral and its OAS3 validations
const { Spectral } = require('@stoplight/spectral')
const { oas3Functions, oas3Rules } = require('@stoplight/spectral/rulesets/oas3')

// Find all OpenAPI files
const fs = require('fs')
const glob = require('glob')

const lint = async () => {
  const fileNames = await new Promise((resolve, reject) => {
    glob('./v2.0/*.openapi.json', {}, (error, match) => {
      if (error) { reject(error) }
      else { resolve(match) }
    })
  })
  
  // Initialize Spectral
  const spectral = new Spectral()
  spectral.addFunctions(oas3Functions())
  spectral.addRules(oas3Rules())
  
  fileNames.forEach(fileName => {
    let specification = JSON.parse(fs.readFileSync(fileName))
    const results = spectral.run(specification)
    // console.dir(results)
  })  
}

lint()
```

**Comment by marbemac** [comment]
Hi @cbetta - great question, and this is something that we could certainly be more clear about (will open a separate PR for readme).

Certain rules (like those param rules you mentioned above) need both the resolved and unresolved document in order to function. Since you don't have any $refs in your document, you should be able to supply it as the resolved version like so:

```js
const results = spectral.run(specification, { resolvedTarget: specification });
```

You can find the run signature here: https://github.com/stoplightio/spectral/blob/next/src/spectral.ts#L21.

I'm now wondering if we shouldn't just default resolvedTarget to the passed in document if resolvedTarget is not explicitly set.

Let me know if the above works for you for now!

**Comment by cbetta** [comment]
Ah yes that works. The documentation wasn't quite up to date on this indeed.

**Comment by tbarn** [comment]
I'm adding a FAQ for this right now.

**Comment by cbetta** [comment]
Did I miss something? I don't see anything added.

**Comment by tbarn** [comment]
PR #89 isn't merged yet, but that should help some. We might be adding one more thing to it.

**Comment by cbetta** [comment]
Ah excellent, just confused cause you said "right now". I hadn't spotted the PR.

**Comment by philsturgeon** [comment]
Hey folks, just to give you an update: the `release/2.0` is getting more mature, so if you would like to check it out your troubles should be over. run now returns a promise, so make sure you add `await`. It is no longer an object with .results, it just an array. 

**Before**

```js
const spectral = new Spectral();
results = spectral.run(obj);
console.log(results.results);
```

**v2.0**

```js
const spectral = new Spectral();
results = await spectral.run(obj);
console.log(results);
```

Please post new issues with any new bugs you find in there, but this is considered done (and will be closed when v2.0 is merged).

**Comment by cbetta** [comment]
Upgraded and all works 👍

**Comment by tbarn** [comment]
@cbetta Glad to hear the improvements worked.
